modules/doc/content/getting_started/installation/install_libtorch.md

```diff
@@ -3,100 +3,208 @@
+using a `libc` version below 2.29 (for `libtorch v 2.1+`) or 2.27 (for `libtorch v 1.8-2.1`)
+or 2.23 (for `libtorch v1.4-1.8`). To check your currently used libc version on +Linux machine+, use the following command:
 
 ```bash
-./configure --with-libtorch=/path/to/my/custom/libtorch
+ldd --version
 ```
 
-Also note that the library files (.so/.dylib) and the headers in the custom
-libtorch installation should follow that of the official distribution handled
-by the `setup_libtorch.sh` script.
+This can be a problem when using the moose conda environment with new versions of libtorch.
 
 !alert-end!
 
-The last step is to compile MOOSE with libtorch support:
+### Using pre-compiled packages from the official website
 
-```bash
-make -j 8
-```
+1. Create and activate a MOOSE environment without pytorch:
+
+   ```bash
+   conda create -n moose moose-dev
+   conda activate moose
+   ```
+
+2. Pull the precompiled libraries using the setup script provided in MOOSE. From the root
+   directory of `moose` this is done by executing the following command:
+
+   ```
+   ./scripts/setup_libtorch.sh --version=2.1
+   ```
+
+3. Configure MOOSE to use libtorch within the root directory of `moose`:
+
+   ```bash
+   ./configure --with-libtorch
+   ```
+
+4. Once moose has been configured to work with libtorch, we need to recompile MOOSE.
+   For testing purposes, we can do the following:
+
+   ```bash
+   cd test
+   make -j 8
+   ./run_test --re=libtorch
+   ```
+
+   If you see tests passing with green colors, the installation was successful.
+
+### Manually installing packages
 
-After configured, compile MOOSE as normal.
+When encountering GLIBC-related compatibility issues on Linux machines the user has two options:
 
-!alert! note
-Due to namespace conflicts between MOOSE and `libtorch`, the folders of source files
-containing libtorch includes need to be excluded from the unit build directories.
-The makefile of the [Stochastic Tools Module](stochastic_tools/stochastic_tools.mk)
-serves as a good example on how to achieve this for applications.
+1. [Rebuilding Petsc and libMesh manually](gcc_install_moose.md) using compatible compilers (most newer systems
+   like Ubuntu come with compatible compilers).
+
+2. Building libtorch from source. The user can find instructions on how to install libtorch
+   from source on the [official website](https://github.com/pytorch/pytorch/blob/master/docs/libtorch.rst).
+
+
+!alert! note title=GPU Support on Linux Workstations
+
+For linux workstations, we recommend [manually building the dependencies of MOOSE](gcc_install_moose.md)
+using suitable system compilers. At the time these instruction are written, only `cuda`-based
+acceleration is tested.
+The following packages need to be also installed to enable this feature:
+
+- [A sutiable Nvidia driver](https://www.nvidia.com/en-us/drivers/)
+- [Cuda toolkit](https://developer.nvidia.com/cuda-toolkit)
```

Comment:
Great question, by `ldd`-ing the library, it links to:
```
libnvToolsExt-847d78f2.so.1 => /opt/libtorch/lib/libnvToolsExt-847d78f2.so.1 (0x00007fe40f38f000)
```
and some other libraries. So I suppose they distribute those libraries together with the package. I'll delete this line here!

Reply:
I suppose they will need it for the manual installation so I'll just leave it there just in case.

Reply:
Which library links to nvToolsExt? I am curious because I never run into issues with the "manual installation" approach without cuda toolkit. Is it possible that MOOSE is not even linking against the said library (which links to nvToolsExt)?

Reply:
```
ldd /opt/libtorch/lib/libtorch_cuda.so 
	linux-vdso.so.1 (0x00007ffe99776000)
	libc10_cuda.so => /opt/libtorch/lib/libc10_cuda.so (0x00007fe40f845000)
	libcudart-9335f6a2.so.12 => /opt/libtorch/lib/libcudart-9335f6a2.so.12 (0x00007fe40f59a000)
	libnvToolsExt-847d78f2.so.1 => /opt/libtorch/lib/libnvToolsExt-847d78f2.so.1 (0x00007fe40f38f000)
	libcudnn.so.8 => /opt/libtorch/lib/libcudnn.so.8 (0x00007fe40f169000)
	libpthread.so.0 => /lib64/libpthread.so.0 (0x00007fe40ef49000)
	libc10.so => /opt/libtorch/lib/libc10.so (0x00007fe40ec89000)
	libtorch_cpu.so => /opt/libtorch/lib/libtorch_cpu.so (0x00007fe3f828b000)
	libdl.so.2 => /lib64/libdl.so.2 (0x00007fe3f8087000)
	librt.so.1 => /lib64/librt.so.1 (0x00007fe3f7e7f000)
	libcublas-37d11411.so.12 => /opt/libtorch/lib/libcublas-37d11411.so.12 (0x00007fe3f15f2000)
	libcublasLt-f97bfc2c.so.12 => /opt/libtorch/lib/libcublasLt-f97bfc2c.so.12 (0x00007fe3cf628000)
	libstdc++.so.6 => /lib64/libstdc++.so.6 (0x00007fe3cf293000)
	libm.so.6 => /lib64/libm.so.6 (0x00007fe3cef11000)
	libgcc_s.so.1 => /lib64/libgcc_s.so.1 (0x00007fe3cecf9000)
	libc.so.6 => /lib64/libc.so.6 (0x00007fe3ce934000)
	/lib64/ld-linux-x86-64.so.2 (0x00007fe46118c000)
	libgomp-52f2fd74.so.1 => /opt/libtorch/lib/libgomp-52f2fd74.so.1 (0x00007fe3ce701000)
```

Reply:
They distribute their own version of the toolkit:
```
ls -l /opt/libtorch/lib                  
total 3999194
-rw-r--r-- 1 germp germp    1461242 Sep 22  2023 libXNNPACK.a
-rw-r--r-- 1 germp germp     754538 Sep 22  2023 libasmjit.a
-rwxr-xr-x 1 germp germp     286536 Sep 22  2023 libbackend_with_compiler.so
-rw-r--r-- 1 germp germp    1059624 Sep 22  2023 libbenchmark.a
-rw-r--r-- 1 germp germp       2456 Sep 22  2023 libbenchmark_main.a
-rwxr-xr-x 1 germp germp     962784 Sep 22  2023 libc10.so
-rwxr-xr-x 1 germp germp    1399368 Sep 22  2023 libc10_cuda.so
-rwxr-xr-x 1 germp germp     839625 Sep 22  2023 libc10d_cuda_test.so
-rwxr-xr-x 1 germp germp      18465 Sep 22  2023 libcaffe2_nvrtc.so
-rw-r--r-- 1 germp germp     561272 Sep 22  2023 libcaffe2_protos.a
-rw-r--r-- 1 germp germp       7258 Sep 22  2023 libclog.a
-rw-r--r-- 1 germp germp      98004 Sep 22  2023 libcpuinfo.a
-rw-r--r-- 1 germp germp      90436 Sep 22  2023 libcpuinfo_internals.a
-rwxr-xr-x 1 germp germp  107496985 Sep 22  2023 libcublas-37d11411.so.12
-rwxr-xr-x 1 germp germp  515090264 Sep 22  2023 libcublasLt-f97bfc2c.so.12
-rwxr-xr-x 1 germp germp     695585 Sep 22  2023 libcudart-9335f6a2.so.12
-rwxr-xr-x 1 germp germp     150200 Sep 22  2023 libcudnn.so.8
-rwxr-xr-x 1 germp germp  125101985 Sep 22  2023 libcudnn_adv_infer.so.8
-rwxr-xr-x 1 germp germp  116220385 Sep 22  2023 libcudnn_adv_train.so.8
-rwxr-xr-x 1 germp germp  633422273 Sep 22  2023 libcudnn_cnn_infer.so.8
-rwxr-xr-x 1 germp germp  132408969 Sep 22  2023 libcudnn_cnn_train.so.8
-rwxr-xr-x 1 germp germp   90555849 Sep 22  2023 libcudnn_ops_infer.so.8
-rwxr-xr-x 1 germp germp   70907289 Sep 22  2023 libcudnn_ops_train.so.8
-rw-r--r-- 1 germp germp  146493704 Sep 22  2023 libdnnl.a
-rw-r--r-- 1 germp germp   23146284 Sep 22  2023 libfbgemm.a
-rw-r--r-- 1 germp germp     266252 Sep 22  2023 libfmt.a
-rw-r--r-- 1 germp germp       3176 Sep 22  2023 libfoxi_loader.a
-rw-r--r-- 1 germp germp    1875854 Sep 22  2023 libgloo.a
-rw-r--r-- 1 germp germp    7315618 Sep 22  2023 libgloo_cuda.a
-rw-r--r-- 1 germp germp     194360 Sep 22  2023 libgmock.a
-rw-r--r-- 1 germp germp       2942 Sep 22  2023 libgmock_main.a
-rw-r--r-- 1 germp germp     205801 Sep 22  2023 libgomp-52f2fd74.so.1
-rw-r--r-- 1 germp germp     853386 Sep 22  2023 libgtest.a
-rw-r--r-- 1 germp germp       3046 Sep 22  2023 libgtest_main.a
-rw-r--r-- 1 germp germp     112040 Sep 22  2023 libittnotify.a
-rwxr-xr-x 1 germp germp     235312 Sep 22  2023 libjitbackend_test.so
-rw-r--r-- 1 germp germp    6563382 Sep 22  2023 libkineto.a
-rwxr-xr-x 1 germp germp     367880 Sep 22  2023 libnnapi_backend.so
-rw-r--r-- 1 germp germp     255140 Sep 22  2023 libnnpack.a
-rw-r--r-- 1 germp germp      20774 Sep 22  2023 libnnpack_reference_layers.a
-rwxr-xr-x 1 germp germp      43681 Sep 22  2023 libnvToolsExt-847d78f2.so.1
-rwxr-xr-x 1 germp germp    9455745 Sep 22  2023 libnvfuser_codegen.so
-rwxr-xr-x 1 germp germp   56875329 Sep 22  2023 libnvrtc-b51b459d.so.12
-rwxr-xr-x 1 germp germp    6846017 Sep 22  2023 libnvrtc-builtins-6c5639ce.so.12.1
-rw-r--r-- 1 germp germp    9415600 Sep 22  2023 libonnx.a
-rw-r--r-- 1 germp germp     619060 Sep 22  2023 libonnx_proto.a
-rw-r--r-- 1 germp germp    1287490 Sep 22  2023 libprotobuf-lite.a
-rw-r--r-- 1 germp germp    6341152 Sep 22  2023 libprotobuf.a
-rw-r--r-- 1 germp germp    6491290 Sep 22  2023 libprotoc.a
-rw-r--r-- 1 germp germp      91900 Sep 22  2023 libpthreadpool.a
-rw-r--r-- 1 germp germp     240146 Sep 22  2023 libpytorch_qnnpack.a
-rw-r--r-- 1 germp germp     159880 Sep 22  2023 libqnnpack.a
-rwxr-xr-x 1 germp germp      41528 Sep 22  2023 libshm.so
-rw-r--r-- 1 germp germp    8589736 Sep 22  2023 libtensorpipe.a
-rw-r--r-- 1 germp germp    3769186 Sep 22  2023 libtensorpipe_cuda.a
-rw-r--r-- 1 germp germp     316668 Sep 22  2023 libtensorpipe_uv.a
-rwxr-xr-x 1 germp germp       7496 Sep 22  2023 libtorch.so
-rwxr-xr-x 1 germp germp  447199673 Sep 22  2023 libtorch_cpu.so
-rwxr-xr-x 1 germp germp 1391025721 Sep 22  2023 libtorch_cuda.so
-rwxr-xr-x 1 germp germp  135218561 Sep 22  2023 libtorch_cuda_linalg.so
-rwxr-xr-x 1 germp germp      20897 Sep 22  2023 libtorch_global_deps.so
-rwxr-xr-x 1 germp germp   22877809 Sep 22  2023 libtorch_python.so
-rwxr-xr-x 1 germp germp     693856 Sep 22  2023 libtorchbind_test.so
-rw-r--r-- 1 germp germp      56448 Sep 22  2023 libunbox_lib.a
```

Reply:
They link to this:
```
/opt/libtorch/lib/libnvToolsExt-847d78f2.so.1
```
they distribute their own versions.

Reply:
But you won't have these if you manually install it from source. Cmake will check for the libraries that come with toolkit.

Reply:
I added a note there that it is only strictly required if they are building libtorch from source.